momento/set_test.go

```diff
@@ -0,0 +1,396 @@
+package momento_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/momentohq/client-sdk-go/utils"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"github.com/momentohq/client-sdk-go/auth"
+	"github.com/momentohq/client-sdk-go/config"
+	. "github.com/momentohq/client-sdk-go/momento"
+)
+
+func HaveLength(length int) types.GomegaMatcher {
+	return WithTransform(
+		func(fetchResp SetFetchResponse) (int, error) {
+			switch rtype := fetchResp.(type) {
+			case *SetFetchHit:
+				return len(rtype.ValueString()), nil
+			default:
+				return 0, fmt.Errorf("expected set fetch hit but got %T", fetchResp)
+			}
+		}, Equal(length),
+	)
+}
+
+func getElements(numElements int) []Value {
+	var elements []Value
+	for i := 0; i < numElements; i++ {
+		elements = append(elements, String(fmt.Sprintf("#%d", i)))
+	}
+	return elements
+}
+
+var _ = Describe("Set methods", func() {
+	var clientProps SimpleCacheClientProps
+	var credentialProvider auth.CredentialProvider
+	var configuration config.Configuration
+	var client SimpleCacheClient
+	var defaultTTL time.Duration
+	var testCacheName string
+	var testSetName string
+	var ctx context.Context
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		credentialProvider, _ = auth.NewEnvMomentoTokenProvider("TEST_AUTH_TOKEN")
+		configuration = config.LatestLaptopConfig()
+		defaultTTL = 3 * time.Second
+
+		clientProps = SimpleCacheClientProps{
+			CredentialProvider: credentialProvider,
+			Configuration:      configuration,
+			DefaultTTL:         defaultTTL,
+		}
+
+		var err error
+		client, err = NewSimpleCacheClient(&clientProps)
+		if err != nil {
+			panic(err)
+		}
+		DeferCleanup(func() { client.Close() })
+
+		testCacheName = uuid.NewString()
+		testSetName = uuid.NewString()
+		Expect(
+			client.CreateCache(ctx, &CreateCacheRequest{CacheName: testCacheName}),
+		).To(BeAssignableToTypeOf(&CreateCacheSuccess{}))
+		DeferCleanup(func() {
+			_, err := client.DeleteCache(ctx, &DeleteCacheRequest{CacheName: testCacheName})
+			if err != nil {
+				panic(err)
+			}
+		})
+	})
+
+	It("errors when the cache is missing", func() {
+		cacheName := uuid.NewString()
+		setName := uuid.NewString()
+
+		fetchResp, err := client.SetFetch(ctx, &SetFetchRequest{
+			CacheName: cacheName,
+			SetName:   setName,
+		})
+		Expect(fetchResp).To(BeNil())
+		Expect(err).To(HaveMomentoErrorCode(NotFoundError))
```

Comment:
I checked and it's safe to write these like so. It will fail if the response is non-nil.

```suggestion
		Expect(
			client.SetFetch(ctx, &SetFetchRequest{
				CacheName: cacheName,
				SetName:   setName,
			}),
		).Error().To(HaveMomentoErrorCode(NotFoundError))
```

Reply:
Cool, I'll switch over to the less verbose form for these.